File: node-js/src/domain/issue.js
class Issue {
    constructor(id, name,description){
        this.id = id;
        this.name = name;
        this.description = description;
    }

    getId() {
        return this.id
    }

    getName() {
        return this.name
    }

    getDescription() {
        return this.description
    }

	update(name, description) {
        return new Issue(
            this.id,
            Issue.isEmpty(name) ? this.name : name,
            Issue.isEmpty(description) ? this.description : description);
    }
    
    static isEmpty(str) {
        return !str || str.length === 0;
    }
}

module.exports = Issue;

File: micronaut-graal/src/main/java/lizzy/medium/compare/micronaut/Issue.java
package lizzy.medium.compare.micronaut;

import io.micronaut.data.annotation.MappedEntity;
import io.micronaut.data.annotation.Id;
import java.util.UUID;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@MappedEntity
@Data
@NoArgsConstructor
@AllArgsConstructor
class Issue {
    @Id
    private UUID id;
    private String name;
    private String description;

    void partialUpdate(Issue partialIssue) {
        if (partialIssue.getName() != null) {
            this.name = partialIssue.getName();
        }

        if (partialIssue.getDescription() != null) {
            this.description = partialIssue.getDescription();
        }
    }
}
